src/ert/shared/models/ensemble_smoother.py

```diff
@@ -24,6 +34,175 @@ def setAnalysisModule(self, module_name: str) -> None:
+                {
+                    "type": identifiers.EVTYPE_EXPERIMENT_STARTED,
+                    "source": f"/ert/experiment/{self.id_}",
+                    "id": str(uuid.uuid1()),
+                }
+            ),
+            prior_context.iteration,
+        )
+
+        self._checkMinimumActiveRealizations(prior_context)
+
+        await loop.run_in_executor(
+            threadpool,
+            self.ert().createRunPath,
+            prior_context,
+        )
+
+        await self._run_hook(
+            HookRuntime.PRE_SIMULATION, prior_context.iteration, loop, threadpool
+        )
+
+        # Post ensemble, parameters, observations to new storage
+        # (nb: this calls self.setPhaseName())
+        ensemble_id = await loop.run_in_executor(threadpool, self._post_ensemble_data)
+
+        experiment_logger.debug("evaluating")
+        await self._evaluate(prior_context, evaluator_server_config)
+
+        # Push simulation results to storage
+        await loop.run_in_executor(threadpool, self._post_ensemble_results, ensemble_id)
+
+        num_successful_realizations = await self.successful_realizations(
+            prior_context.iteration
+        )
+
+        self.checkHaveSufficientRealizations(num_successful_realizations)
+
+        await self._run_hook(
+            HookRuntime.POST_SIMULATION, prior_context.iteration, loop, threadpool
+        )
+
+        await self._run_hook(
+            HookRuntime.PRE_FIRST_UPDATE, prior_context.iteration, loop, threadpool
+        )
+
+        await self._run_hook(
+            HookRuntime.PRE_UPDATE, prior_context.iteration, loop, threadpool
+        )
+
+        await self.dispatch(
+            CloudEvent(
+                {
+                    "type": identifiers.EVTYPE_EXPERIMENT_ANALYSIS_STARTED,
+                    "source": f"/ert/experiment/{self.id_}",
+                    "id": str(uuid.uuid1()),
+                }
+            ),
+            prior_context.iteration,
+        )
+
+        experiment_logger.debug("running update...")
+        try:
+            await loop.run_in_executor(
+                threadpool, self.facade.smoother_update, prior_context
+            )
+        except ErtAnalysisError as e:
+            experiment_logger.exception("analysis failed")
+            await self.dispatch(
+                CloudEvent(
+                    {
+                        "type": identifiers.EVTYPE_EXPERIMENT_FAILED,
+                        "source": f"/ert/experiment/{self.id_}",
+                        "id": str(uuid.uuid1()),
+                    },
+                    {
+                        "error": str(e),
+                    },
+                ),
+                prior_context.iteration,
+            )
+            raise ErtRunError(
+                f"Analysis of simulation failed with the following error: {e}"
+            ) from e
+
+        experiment_logger.debug("update complete")
+
+        await self.dispatch(
```

Comment:
We might consider creating a helper class for `dispatch`, which automatically inject the correct source string.

Reply:
Good idea, implemented. Unsure of the name I chose,   "_dispatch_ee"